### DISCUSSION TITLE
run() in loop?

### DISCUSSION THREAD

**Discussion by HamzaHajeir** [khoih-prog/ESP8266TimerInterrupt#25] [Q&A]
Hi there

I'm reading out the library and discovering how does it work, preparing to use it in the next project.

However, I"ve read here and there that run() must be called in loop() to work:
>    // this function must be called inside loop()
>    void IRAM_ATTR run();

but no one example calls run(), also couldn't find where the library attaches to esp8266 hardware timer to run its timing cababilities ..?

Best,
Hamza

**Comment by khoih-prog** [comment]
Normally, it's easier and faster to scan through and try the examples to master the usage of a library.

As in 

https://github.com/khoih-prog/ESP8266TimerInterrupt/blob/8a003893199a999b4c3ef7e56a86f306ab08937b/examples/ISR_16_Timers_Array/ISR_16_Timers_Array.ino#L99

of example [**ISR_16_Timers_Array**](https://github.com/khoih-prog/ESP8266TimerInterrupt/tree/master/examples/ISR_16_Timers_Array)

**Comment by HamzaHajeir** [comment]
I've checked out examples, but now with a second eye to this example I
assume ESP8266Timer is the external class provided by the core, correct?


BR,
Hamza
